### PR TITLE
pkp/pkp-lib#1472 Fix galley/galley file/representation cardinalities

### PR DESCRIPTION
**Pull request by asmecher**

### CODE REVIEW

File: controllers/grid/articleGalleys/ArticleGalleyGridCellProvider.inc.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * @file controllers/grid/queries/ArticleGalleyGridCellProvider.inc.php
```
Comment (bozana) [inline controllers/grid/articleGalleys/ArticleGalleyGridCellProvider.inc.php]: wrong path :-)

File: controllers/grid/articleGalleys/ArticleGalleyGridCellProvider.inc.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+ *
+ * Copyright (c) 2016 Simon Fraser University Library
+ * Copyright (c) 2000-2016 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class ArticleGalleyGridCellProvider
+ * @ingroup controllers_grid_users_author
```
Comment (bozana) [inline controllers/grid/articleGalleys/ArticleGalleyGridCellProvider.inc.php]: this group?

File: controllers/grid/articleGalleys/ArticleGalleyGridCellProvider.inc.php [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+class ArticleGalleyGridCellProvider extends DataObjectGridCellProvider {
+
+	/** @var Submission **/
+	var $_submission;
+
+	/**
+	 * Constructor
+	 * @param $submission Submission
+	 * @param $stageId int
+	 * @param $queriesAccessHelper QueriesAccessHelper
```
Comment (bozana) [inline controllers/grid/articleGalleys/ArticleGalleyGridCellProvider.inc.php]: params to correct

File: controllers/grid/articleGalleys/ArticleGalleyGridCellProvider.inc.php [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+	// Template methods from GridCellProvider
+	//
+	/**
+	 * Extracts variables for a given column from a data element
+	 * so that they may be assigned to template before rendering.
+	 * @param $row GridRow
+	 * @param $column GridColumn
+	 * @return array
+	 */
+	function getTemplateVarsFromRowColumn($row, $column) {
```
Comment (bozana) [inline controllers/grid/articleGalleys/ArticleGalleyGridCellProvider.inc.php]: copydoc?

File: controllers/grid/articleGalleys/ArticleGalleyGridRow.inc.php [hunk 5]
```diff
@@ -0,0 +1,110 @@
+<?php
+
+/**
+ * @file controllers/grid/queries/ArticleGalleyGridRow.inc.php
```
Comment (bozana) [inline controllers/grid/articleGalleys/ArticleGalleyGridRow.inc.php]: path

File: controllers/grid/articleGalleys/ArticleGalleyGridRow.inc.php [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+ *
+ * Copyright (c) 2016 Simon Fraser University Library
+ * Copyright (c) 2000-2016 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class ArticleGalleyGridRow
+ * @ingroup controllers_grid_queries
```
Comment (bozana) [inline controllers/grid/articleGalleys/ArticleGalleyGridRow.inc.php]: group?

File: classes/issue/IssueGalley.inc.php [hunk 7]
```diff
@@ -190,6 +193,19 @@ function getBestGalleyId(&$journal) {
 		}
 		return $this->getId();
 	}
+
+	/**
+	 * Get the submission file corresponding to this galley.
+	 * @return SubmissionFile
```
Comment (bozana) [inline classes/issue/IssueGalley.inc.php]: IssueFile or PKPFile

File: plugins/generic/downloadableFileArticleGalley/locale/en_US/locale.xml [hunk 8]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE locale SYSTEM "../../../../../lib/pkp/dtd/locale.dtd">
+
+<!--
+  * plugins/generic/downloadableFileArticleGalley/locale/en_US/locale.xml
+  *
+  * Copyright (c) 2014-2016 Simon Fraser University Library
```
Comment (bozana) [inline plugins/generic/downloadableFileArticleGalley/locale/en_US/locale.xml]: Just 2016?

File: plugins/generic/htmlArticleGalley/HtmlArticleGalleyPlugin.inc.php [hunk 9]
```diff
@@ -93,53 +96,26 @@ function headerCallback($hookName, $args) {
...
-			if ($file) {
-				$fileId = $file->getFileId();
-			} else {
-				assert(false); // No HTML file in this HTML Article galley?
-				return false;
-			}
-		}
-		$contents = $fileManager->readFile($fileId);
+		$contents = $fileManager->readFile($galley->getFile()->getFileId());
+print_r($galley->getFile()->getFileId());
```
Comment (bozana) [inline plugins/generic/htmlArticleGalley/HtmlArticleGalleyPlugin.inc.php]: this can be removed?

Comment (asmecher) [inline plugins/generic/htmlArticleGalley/HtmlArticleGalleyPlugin.inc.php]: This plugin still needs a lot of rewriting -- I'll add it to the list and update the issue entry. I'm unlikely to get to this over the next few days. It'll need e.g. was to associate image files and CSS files with a galley, which none of the other formats have required.

File: plugins/generic/downloadableFileArticleGalley/locale/en_US/locale.xml (hunk 8 above)
Comment (asmecher) [inline plugins/generic/downloadableFileArticleGalley/locale/en_US/locale.xml]: I probably borrowed this from some other source file. Not a big deal.

**Pull request merged**